ui/src/shared/components/ResizeContainer.js

```diff
@@ -1,47 +1,46 @@
-import React, {PropTypes} from 'react'
-import ResizeHandle from 'shared/components/ResizeHandle'
+import React, {Component, PropTypes} from 'react'
 import classnames from 'classnames'
 
-const {node, number, string} = PropTypes
+import ResizeHandle from 'shared/components/ResizeHandle'
 
 const maximumNumChildren = 2
 const minimumTopHeight = 200
 const minimumBottomHeight = 200
+const initialHeight = '50%'
 
-const ResizeContainer = React.createClass({
-  propTypes: {
-    children: node.isRequired,
-    containerClass: string.isRequired,
-    minTopHeight: number,
-    minBottomHeight: number,
-  },
-
-  getDefaultProps() {
-    return {
-      minTopHeight: minimumTopHeight,
-      minBottomHeight: minimumBottomHeight,
-    }
-  },
-
-  getInitialState() {
-    return {
-      topHeight: '60%',
-      bottomHeight: '40%',
+class ResizeContainer extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
       isDragging: false,
+      topHeight: this.props.initialTopHeight,
+      bottomHeight: this.props.initialBottomHeight,
     }
-  },
 
-  handleStopDrag() {
-    this.setState({isDragging: false})
-  },
+    this.handleStartDrag = ::this.handleStartDrag
+    this.handleStopDrag = ::this.handleStopDrag
+    this.handleMouseLeave = ::this.handleMouseLeave
+    this.handleDrag = ::this.handleDrag
+  }
+
+  static defaultProps = {
+    minTopHeight: minimumTopHeight,
+    minBottomHeight: minimumBottomHeight,
+    initialTopHeight: initialHeight,
```

Comment:
it's a bit confusing to assign `initialHeight` to both `initialTopHeight` and `initialBottomHeight` -- maybe rename this constant `fiftyPercent`?